grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/AbstractComputeQuery.java

```diff
@@ -117,30 +125,35 @@ void initSubGraph() {
         keySpace = theGraph.getKeyspace();
         url = theGraph.admin().getEngineUrl();
 
+        EntityType metaEntityType = theGraph.admin().getMetaEntityType();
+        entityTypes.addAll(metaEntityType.subs());
+        entityTypes.remove(metaEntityType);
+
+        ResourceType<?> metaResourceType = theGraph.admin().getMetaResourceType();
+        resourceTypes.addAll(metaResourceType.subs());
+        resourceTypes.remove(metaResourceType);
+
+        RelationType metaRelationType = theGraph.admin().getMetaRelationType();
+        relationTypes.addAll(metaRelationType.subs());
+        relationTypes.remove(metaRelationType);
```

Comment:
If we do this here, doesn't it mean we end up doing a lot of extra work if `subLabels` is not empty?